### DISCUSSION TITLE
Switching from module import to .test.ts causes processTicksAndRejections

### DISCUSSION THREAD

**Discussion by ollyde** [avajs/ava#2728] [Q&A]
Node v15.11.0
Using Supertest
TypeScript
M1 Macbook Pro

I was importing the modules into the test.ts and it was working fine.

I decided to use [test-name].test.ts and I started getting this error:

```
  Server.setupListenHandle [as _listen2] (node:net:1310:16)
  listenInCluster (node:net:1358:12)
  Server.listen (node:net:1445:7)
  Function.listen (node_modules/express/lib/application.js:618:24)
  index.ts:41:7
  fulfilled (index.ts:5:58)
  processTicksAndRejections (node:internal/process/task_queues:94:5)
```

I found that the temporary fix was ```-concurrency 1``` but that makes it super slow.

My AVA config is as such

```
"ava": {
    "babel": false,
    "compileEnhancements": false,
    "extensions": [
      "ts"
    ],
    "require": [
      "ts-node/register"
    ],
    "files": [
      "src/tests/*"
    ]
},
```
Not sure what the fix is here; to do it 'properly'. I will need to import the tests again.

**Comment by novemberborn** [comment]
Something fails, yes, but critically you haven't shared the error message itself. Or code.

Judging from the configuration you're using AVA 2? We can't help you there, AVA 3 is the latest stable version.

**Reply by ollyde** [reply]
@novemberborn the code is 10,000s of line. We updated to 3.15.0 and we were seeing the same issue across our TypeScript projects; testing in JavaScript **feels** very backwards 😂

Is there any reason why this was moved to discussion and out of the bugs area, since it's a bug and not really a discussion? And it will never get looked at now.

The issue is as explained. Doing tests with [some-file-name].test.ts vs just importing it into the root test causes this error.

**Reply by novemberborn** [reply]
> The issue is as explained. Doing tests with [some-file-name].test.ts vs just importing it into the root test causes this error.

But you haven't shared the error, only a stack trace of Node.js internals and something from your web server.

AVA just loads the files, there's nothing special about the filename. You also have `ts-node/register` involved. Have you had a look recently at our [TypeScript recipe](https://github.com/avajs/ava/blob/main/docs/recipes/typescript.md)? There's a couple options available.



> Is there any reason why this was moved to discussion and out of the bugs area, since it's a bug and not really a discussion? And it will never get looked at now.

Just because you're encountering an error in your project does not mean there's a bug in AVA. At this stage it's more suitable for a discussion. If there's a bug we can move it back or open a new issue.

And here I am, engaging with you, on a Saturday no less.

**Reply by ollyde** [reply]
@novemberborn that was the only error.

"AVA just loads the files, there's nothing special about the filename. You also have ts-node/register involved. Have you had a look recently at our TypeScript recipe? There's a couple options available."

This is not true. When switching from module import to .test.ts it causes the error. I can see that repeated.

I cba with this. If you don't want to help and just move my projects from the bug list then just delete this issue please.
